monitoring/bin/deploy_monitoring_cluster.sh

```diff
@@ -65,10 +65,37 @@ fi
 # Elasticsearch Datasource for Grafana
 ELASTICSEARCH_DATASOURCE="${ELASTICSEARCH_DATASOURCE:-false}"
 if [ "$ELASTICSEARCH_DATASOURCE" == "true" ]; then
-  log_verbose "Provisioning Elasticsearch datasource for Grafana"
-  kubectl delete secret -n $MON_NS --ignore-not-found grafana-datasource-es
-  kubectl create secret generic -n $MON_NS grafana-datasource-es --from-file monitoring/grafana-datasource-es.yaml
-  kubectl label secret -n $MON_NS grafana-datasource-es grafana_datasource=1 sas.com/monitoring-base=kube-viya-monitoring
+
+  # Check to see if logging components have already been deployed.
+  if [[ $(kubectl get pods -A -l app.kubernetes.io/instance=v4m-fb -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+    log_verbose "No logging components found.  Skipping Elasticsearch data source set up.";
+  elif [[ $(kubectl get pods -A -l app.kubernetes.io/instance=v4m-fb -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) > 1 ]]; then
+    log_verbose "Too many logging components found.  Skipping Elasticsearch data source set up.";
+  else
+    # Replace logging namespace name and password in grafana-datasource-es.yaml file
+    logNS=$(kubectl get pods -A -l app.kubernetes.io/instance=v4m-fb -o custom-columns=:metadata.namespace --no-headers | uniq)
```

Comment:
Hmmm, we use the LOG_NS to identify the "logging" namespace in our deployment scripts.  I wonder if we should use it here as well.  Perhaps the logic should be to use LOG_NS (if set) and then go looking for stuff if it isn't set.  

Also, how comfortable are we that the uniq command will be available?

Reply:
In this case, we can just get the actual namespace instead of asking for it as input. It's either there or not, so I'm not sure of the value of an override here.

To my knowledge, I believe `uniq` is always available.

Reply:
I'm less concerned about asking for it but if  LOG_NS has been set in the USER_DIR, it seems we should be using that rather than looking around for the logging components.  I also worry about similarly named variables that serve similar purposes (LOG_NS vs. logNS) causing confusion even if they exist in different scripts.

Reply:
If we assume that both MON_NS and LOG_NS are being appropriately set in their USER_DIR, I can go ahead and explicitly use them.

The only reason why I went with the "search for a pod" approach is because LOG_NS is only declared in the logging/common.sh script (which may not necessarily get called).  In theory, that could be fixed if we start calling logging/common.sh in monitoring scripts, but I wasn't sure if we wanted to go that route (yet).

Reply:
I still don't get why we would explicitly use whatever `LOG_NS` is set to when we have to check for the existence of the actual logging deployment there anyway. Why not just get the real namespace instead of setting ourselves up for a mismatch situation?

Reply:
For now, let's leave this as-is.  However, if/when we move to a stand-alone script, we should think about how we would handle the situation where the USER_DIR includes setting LOG_NS to a namespace but we do not find an instance of the logging components deployed there.  It seems like something is amiss in that case and the user should be asked to resolve the discrepancy.  If we have settings that can be interpreted as "global settings" (e.g. LOG_NS, MON_NS), it seems like we should honor them.